packages/api-client-core/src/support.ts

```diff
@@ -19,38 +19,78 @@ export class GadgetInternalError extends Error {}
-    super(incomingMessage.startsWith("GGT_") ? incomingMessage : `${errorCode}: ${incomingMessage}`);
+  constructor(incomingMessage: string, readonly code: string) {
+    super(incomingMessage.startsWith("GGT_") ? incomingMessage : `${code}: ${incomingMessage}`);
   }
 }
 
 /**
- * Interface representing one message on one invalid field for a `GadgetValidationError`
+ * Interface representing one message on one invalid field for a `InvalidRecordError`
  */
 export interface InvalidFieldError {
+  /** Which field of a record this error is for */
   apiIdentifier: string;
+  /** Human facing string representing the error */
   message: string;
 }
 
 /**
  * A Gadget API error representing a backend validation error on the input data for an action. Thrown when any of the validations configured on a model fail for the given input data. Has a `validationErrors` property describing which fields failed validation, with messages for each.
  **/
-export class GadgetValidationError extends Error {
-  errorCode = "GGT_INVALID_RECORD";
-
-  constructor(message: string, readonly validationErrors: InvalidFieldError[]) {
-    super(message);
+export class InvalidRecordError extends Error {
+  code = "GGT_INVALID_RECORD";
+  name = "InvalidRecordError";
+
+  /** @private */
+  statusCode = 422;
+  /** @private */
+  causedByClient = true;
+  /**
+   * A list of validation errors for each field that failed validation.
+   */
+  readonly validationErrors: InvalidFieldError[];
+  /**
+   * The API identifier of the model for this record which failed to validate
+   */
+  readonly modelApiIdentifier?: string;
+  /**
+   * The record that failed to validate, if available
+   */
+  readonly record?: Record<string, any>;
+
+  constructor(message: string | null, validationErrors: InvalidFieldError[], modelApiIdentifier?: string, record?: Record<string, any>) {
+    super(
+      message ??
+        `GGT_INVALID_RECORD: ${modelApiIdentifier ?? "Record"} is invalid and can't be saved. ${validationErrors
+          .slice(0, 3)
+          .map(({ apiIdentifier, message }) => `${apiIdentifier} ${message}`)
+          .join(", ")}${
+          validationErrors.length > 3
+            ? `, and ${validationErrors.length - 3} more error${validationErrors.length > 4 ? "s" : ""} need${
+                validationErrors.length > 4 ? "" : "s"
+              } to be corrected`
+            : ""
+        }.`
+    );
+    this.validationErrors = validationErrors;
+    this.modelApiIdentifier = modelApiIdentifier;
+    this.record = record;
   }
 }
 
 /**
- * A Gadget API error that represents an error from the server. Thrown when the server enounters data that is not unique despite the existence of unique validation on a field. If you receive this error, it is likely that you added a unique validation to a field that has duplicate data.
+ * @deprecated Use `InvalidRecordError` instead, here for backwards compatability
+ */
+export const GadgetValidationError = InvalidRecordError;
```

Comment:
nit: Any reason to not continue using `GadgetValidationError`?

Reply:
Was trying to get it to match the `GGT_` error code, which is `GGT_INVALID_RECORD` ?

Reply:
Ah, makes sense. Do you think (this PR, or another, up to you) that we should do that for all of the other errors? For example, `NonUniqueDataError` vs `GadgetNonUniqueDataError`.

Reply:
Ya I do, I will follow up